Sources/AsyncHTTPClient/SSLContextCache.swift

```diff
@@ -88,17 +79,20 @@ extension SSLContextCache {
 
         logger.debug("creating new SSL context",
                      metadata: ["ahc-tls-config": "\(tlsConfiguration)"])
-        let newSSLContext = self.threadPool.runIfActive(eventLoop: eventLoop) {
-            try NIOSSLContext(configuration: tlsConfiguration)
+        let promise = eventLoop.makePromise(of: NIOSSLContext.self)
+        self.threadPool.async {
```

Comment:
I think NIO has API to do this for you: `threadPool.asyncWithFuture`

Reply:
It's not a threadPool :P The name tricked you.

Reply:
I kept the `threadPool` name to trick you into thinking I thought it was a `NIOThreadPool` ;) 

https://github.com/apple/swift-nio/blob/d161bf658780b209c185994528e7e24376cf7283/Sources/NIO/DispatchQueue%2BWithFuture.swift#L17-L34